inspirehep/modules/literaturesuggest/tasks.py

```diff
@@ -24,205 +24,220 @@
-            if "Citeable" not in collections:
-                data['collections'].append({'primary': "Citeable"})
-    # ============================
-    # Title source and cleanup
-    # ============================
-    try:
-        # Clean up all extra spaces in title
-        data['titles'][0]['title'] = " ".join(
-            data['titles'][0]['title'].split()
+    builder = LiteratureBuilder(source='submitter')
+
+    builder.add_authors(
+        authors=form_fields.get('authors', []),
+        supervisors=form_fields.get('supervisors')
+    )
+    builder.add_title(title=form_fields.get('title', []))
+
+    document_type = 'conference paper' if form_fields.get('conf_name') \
+        else form_fields.get('type_of_doc', [])
+    builder.add_document_type(document_type=document_type)
+
+    if form_fields.get('abstract'):
+        builder.add_abstract(
+            abstract=form_fields.get('abstract'),
+            source='arXiv' if form_fields.get('categories') else None
         )
-        title = data['titles'][0]['title']
-    except (KeyError, IndexError):
-        title = ""
-    if form_fields.get('title_arXiv'):
-        title_arxiv = " ".join(form_fields.get('title_arXiv').split())
-        if title == title_arxiv:
-            data['titles'][0]["source"] = "arXiv"
-        else:
-            data['titles'].append({
-                'title': title_arxiv,
-                'source': "arXiv"
-            })
-    if form_fields.get('title_crossref'):
-        title_crossref = " ".join(
-            form_fields.get('title_crossref').split()
+
+    if form_fields.get('arxiv_id') and form_fields.get('categories'):
+        arxiv_categories = form_fields.get('categories').split()
+        arxiv_id = form_fields.get('arxiv_id')
+        if len(arxiv_id.split('/')) == 2:
+            if arxiv_id.split('/')[0] not in arxiv_categories:
+                arxiv_categories.append(arxiv_id.split('/')[0])
+
+        builder.add_arxiv_eprint(
+            arxiv_id=form_fields.get('arxiv_id'),
+            arxiv_categories=arxiv_categories
+        )
+
+    if form_fields.get('doi'):
+        builder.add_doi(doi=form_fields.get('doi'))
+
+    if form_fields.get('subject_term'):
+        builder.add_inspire_categories(
+            subject_terms=form_fields.get('subject_term'),
+            source='user'
         )
-        if title == title_crossref:
-            data['titles'][0]["source"] = "CrossRef"
-        else:
-            data['titles'].append({
-                'title': title_crossref,
-                'source': "CrossRef"
-            })
+
+    if form_fields.get('extra_comments'):
+        builder.add_private_notes(form_fields.get('extra_comments'))
+
+    if form_fields.get('nonpublic_note'):
+        builder.add_private_notes(form_fields.get('nonpublic_note'))
+
+    if form_fields.get('hidden_notes'):
+        builder.add_private_notes(form_fields.get('hidden_notes'))
```

Comment:
Where do all these notes come from? In the form, I only see 'additional comments' and 'references' that should become notes.